### DISCUSSION TITLE
Feature suggestion, embeded content from links.

### DISCUSSION THREAD

**Discussion by coldwinds** [zim-desktop-wiki/zim-desktop-wiki#2070] [Ideas]
Hello,

Since we can put links of some notes into one note, could we choose to also embed the content of the links into the note (or quick content preview of a link)? like iframe of html, or quote directly from some links's content.

**Comment by pfalcon** [comment]
Yes, the latter. It's called "clipboard". Copy needed content from the old note, paste into the new note. Voila.

**Comment by coldwinds** [comment]
> Yes, the latter. It's called "clipboard". Copy needed content from the old note, paste into the new note. Voila.

Will the pasted content update automatically when the linked note get updated? If it's pasted, not embeded.

**Comment by pfalcon** [comment]
No, but you can update it yourself if just linking to is not enough, and it can be used right away, without waiting for someone to implement it for you (potentially very long), or implementing it yourself (potentially, spending a lot of effort on that, which in the end may not pay off).

**Comment by quesada** [comment]
This would be lovely for long-form writing. If you write say a book, and have separate notes per chapter, but want to, at some point, see or export the entire book as a single note. I see the use case

**Comment by Aikatsui** [comment]
This feature has on onenote. No any FOSS tool has come close to OneNote

**Comment by undoingtech** [comment]
I would like this feature. OP calls it an "iframe". When I think about it, the word "snippet" comes to mind.

I write a lot of how-to's. How to setup nextcloud/mailinabox/whatever-software on linode. How to give a user sudo access. How to use ssh. It would be nice to only once write "How to setup an ssh user", and then see that automatically as a "snippet" or "in an iframe" in my longer consolidated How-to's. Bonus points if all the snippets automatically update after an edit. I wouldn't have to worry as much about a How-to having old instructions because I forgot to copy paste one part.

But I get also get by with links for now.

**Comment by candideu** [comment]
I believe this is also called transclusion. Has anyone figured this out?

**Reply by introt** [reply]
It hasn't been implemented yet. See also #1818

**Comment by Lost-Projects** [comment]
I'm commenting here because this is pretty close to something I was going to propose - in fact , it may be the same thing.

The idea is: Allow inline "quotes" of other page sections (as marked by anchor tags). A couple of use examples I thought of were of an economic researcher who might have a page for each of a number of indicators (e.g inflation, interest rates, productivity etc.). Each page would be a report that would contain a figure and a conclusion. If the researcher wanted to then produce a report on the state of the whole economy, having a page that quoted the relevant figures and conclusions from the other pages would be useful - especially if those sections were updated when the page was opened/reloaded/rendered.

Another, weaker use case was of book chapter pages collated into a "whole book" page as mentioned above. Changes to individual chapters could then be read in context to check for errors/contradictions.

My use case would be to have "dashboards" to monitor projects. Each project would have a summary section and as work is done, the updated summary is then included in a "summary of summaries" page.

My guess (not a coder) for a mechanism would be to enhance the anchor tag or include it in a quote or (as above) "snippet" tag which would have open and close forms to define the area both in the originating page and the quoting page. This would be to limit the quote/snippet in the original page and define it's position in the quoting page. This would be done in the manner of a search/replace function using tags as regular expressions. Possibly mechanisms like the ToC or the task list work in the same way.

Given that wiki pages are static, my proposal is that page areas that are quotes are copy/pasted when the page is opened or refreshed and rendered. I'm NOT proposing some kind of "stock market ticker" thing (like a dynamic webpage), which would anyway likely involve remote links.

One possible problem might be quotes within quotes, which might either slow things down or create an infinite loop. It may not be a problem (aside from rendering time), but it might be necessary to limit the number of nested quotes or refuse to implement "circular" quoting where a page A quotes page B that is already quoting page A etc. A simple rule might be to never quote the same tag twice in one "loop" and never quote the originating tag content in the output.

Some things that seem similar:
https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/39
https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/42
https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/269

I'll look at the custom tool in the comment above, but I'll have to read more closely to understand how it works.
